{
  "uid" : "c6cf531b86d04bea",
  "name" : "test_insufficient_approval",
  "fullName" : "tests.brownie.token.tests.test_transferFrom#test_insufficient_approval",
  "historyId" : "0ae63a08126a8f0616f222dcfe6ba70d",
  "time" : {
    "start" : 1641982479214,
    "stop" : 1641982479214,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
  "statusTrace" : "brownie_project = <Project 'TokenProject'>\naccounts = <brownie.network.account.Accounts object at 0x7f3a771bb460>\n\n    @pytest.fixture(scope=\"module\")\n    def token(brownie_project, accounts):\n>       return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n\ntests/brownie/token/tests/conftest.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py:600: in __call__\n    return tx[\"from\"].deploy(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:508: in deploy\n    receipt, exc = self._make_transaction(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:725: in _make_transaction\n    raise VirtualMachineError(e) from None\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = VirtualMachineError(ValueError(\"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries ex...ion refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"))\nexc = \"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by N...ction refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"\n\n    def __init__(self, exc: ValueError) -> None:\n        try:\n            exc = exc.args[0]\n        except Exception:\n            pass\n    \n        if isinstance(exc, dict) and \"message\" in exc:\n            if \"data\" not in exc:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.message: str = exc[\"message\"].rstrip(\".\")\n    \n            if isinstance(exc[\"data\"], str):\n                # handle parity exceptions - this logic probably is not perfect\n                if \"0x08c379a0\" in exc[\"data\"]:\n                    revert_type, err_msg = [i.strip() for i in exc[\"data\"].split(\"0x08c379a0\", 1)]\n                    err_msg = eth_abi.decode_abi([\"string\"], HexBytes(err_msg))\n                    err_msg = f\"{revert_type} '{err_msg}'\"\n                elif exc[\"data\"].endswith(\"0x\"):\n                    err_msg = exc[\"data\"][:-2].strip()\n                else:\n                    err_msg = exc[\"data\"]\n                raise ValueError(f\"{self.message}: {err_msg}\") from None\n    \n            try:\n                txid, data = next((k, v) for k, v in exc[\"data\"].items() if k.startswith(\"0x\"))\n            except StopIteration:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.txid: str = txid\n            self.source: str = \"\"\n            self.revert_type: str = data[\"error\"]\n            self.pc: Optional[str] = data.get(\"program_counter\")\n            if self.pc and self.revert_type == \"revert\":\n                self.pc -= 1\n    \n            self.revert_msg: Optional[str] = data.get(\"reason\")\n            self.dev_revert_msg = brownie.project.build._get_dev_revert(self.pc)\n            if self.revert_msg is None and self.revert_type in (\"revert\", \"invalid opcode\"):\n                self.revert_msg = self.dev_revert_msg\n            elif self.revert_msg == \"Failed assertion\":\n                self.revert_msg = self.dev_revert_msg or self.revert_msg\n    \n        else:\n>           raise ValueError(str(exc)) from None\nE           ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py:121: ValueError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "configure_neon_network",
    "time" : {
      "start" : 1641982460018,
      "stop" : 1641982460018,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "brownie_project",
    "time" : {
      "start" : 1641982460753,
      "stop" : 1641982475591,
      "duration" : 14838
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "setup_env",
    "time" : {
      "start" : 1641982475979,
      "stop" : 1641982475979,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "accounts",
    "time" : {
      "start" : 1641982475979,
      "stop" : 1641982476248,
      "duration" : 269
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "token",
    "time" : {
      "start" : 1641982478167,
      "stop" : 1641982478374,
      "duration" : 207
    },
    "status" : "broken",
    "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n",
    "statusTrace" : "  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 1126, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 932, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/neon-compatibility/neon-compatibility/tests/brownie/token/tests/conftest.py\", line 49, in token\n    return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py\", line 600, in __call__\n    return tx[\"from\"].deploy(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 508, in deploy\n    receipt, exc = self._make_transaction(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 725, in _make_transaction\n    raise VirtualMachineError(e) from None\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py\", line 121, in __init__\n    raise ValueError(str(exc)) from None\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  } ],
  "afterStages" : [ {
    "name" : "accounts::0",
    "time" : {
      "start" : 1641982505103,
      "stop" : 1641982505103,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Brownie"
  }, {
    "name" : "parentSuite",
    "value" : "tests.brownie.token.tests"
  }, {
    "name" : "suite",
    "value" : "test_transferFrom"
  }, {
    "name" : "host",
    "value" : "fv-az28-122"
  }, {
    "name" : "thread",
    "value" : "3612-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.brownie.token.tests.test_transferFrom"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 77,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 77
      },
      "items" : [ {
        "uid" : "686592d6fd201afd",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/460//#testresult/686592d6fd201afd",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189460db20>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641980268252,
          "stop" : 1641980268252,
          "duration" : 0
        }
      }, {
        "uid" : "ec4566601bb4db29",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/459//#testresult/ec4566601bb4db29",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641975867958,
          "stop" : 1641975868158,
          "duration" : 200
        }
      }, {
        "uid" : "7732d7c5f933e6b1",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/458//#testresult/7732d7c5f933e6b1",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189a842400>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974828339,
          "stop" : 1641974828339,
          "duration" : 0
        }
      }, {
        "uid" : "3886ea643d3e8b50",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/457//#testresult/3886ea643d3e8b50",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1896042940>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974206933,
          "stop" : 1641974206933,
          "duration" : 0
        }
      }, {
        "uid" : "6ccfa83fd1c31aa1",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/456//#testresult/6ccfa83fd1c31aa1",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641958013275,
          "stop" : 1641958013275,
          "duration" : 0
        }
      }, {
        "uid" : "b8421c1a1e401944",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/455//#testresult/b8421c1a1e401944",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641952209300,
          "stop" : 1641952209300,
          "duration" : 0
        }
      }, {
        "uid" : "90fc132d0c3a3a45",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/454//#testresult/90fc132d0c3a3a45",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641950686943,
          "stop" : 1641950686943,
          "duration" : 0
        }
      }, {
        "uid" : "138eb57d88396da",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/453//#testresult/138eb57d88396da",
        "status" : "broken",
        "statusDetails" : "ValueError: ['Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument']",
        "time" : {
          "start" : 1641947815957,
          "stop" : 1641947815957,
          "duration" : 0
        }
      }, {
        "uid" : "fdfb2de59091f491",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/452//#testresult/fdfb2de59091f491",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641871098999,
          "stop" : 1641871099151,
          "duration" : 152
        }
      }, {
        "uid" : "8b5b5afc61333214",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/451//#testresult/8b5b5afc61333214",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641861492971,
          "stop" : 1641861493175,
          "duration" : 204
        }
      }, {
        "uid" : "1a264fee0e07a699",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/450//#testresult/1a264fee0e07a699",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641784607373,
          "stop" : 1641784607583,
          "duration" : 210
        }
      }, {
        "uid" : "cb7f12ab8ecfaaae",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/449//#testresult/cb7f12ab8ecfaaae",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641774926548,
          "stop" : 1641774926703,
          "duration" : 155
        }
      }, {
        "uid" : "8c4de1095459a161",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/448//#testresult/8c4de1095459a161",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641698381267,
          "stop" : 1641698381412,
          "duration" : 145
        }
      }, {
        "uid" : "feb0cdb0aa0d7c79",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/447//#testresult/feb0cdb0aa0d7c79",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641688892108,
          "stop" : 1641688892323,
          "duration" : 215
        }
      }, {
        "uid" : "1f3369fd0874a330",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/446//#testresult/1f3369fd0874a330",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641612001804,
          "stop" : 1641612002008,
          "duration" : 204
        }
      }, {
        "uid" : "46f6a59a01a0c946",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/445//#testresult/46f6a59a01a0c946",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641602285039,
          "stop" : 1641602285288,
          "duration" : 249
        }
      }, {
        "uid" : "8e9f985843b3b336",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/441//#testresult/8e9f985843b3b336",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641525625822,
          "stop" : 1641525626043,
          "duration" : 221
        }
      }, {
        "uid" : "d1139a633af7ab2b",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/440//#testresult/d1139a633af7ab2b",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641515822952,
          "stop" : 1641515823107,
          "duration" : 155
        }
      }, {
        "uid" : "74ba63ce1f4c366e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/434//#testresult/74ba63ce1f4c366e",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641439121351,
          "stop" : 1641439121581,
          "duration" : 230
        }
      }, {
        "uid" : "6f14a1de0c2a09aa",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/432//#testresult/6f14a1de0c2a09aa",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641429469734,
          "stop" : 1641429469954,
          "duration" : 220
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "c6cf531b86d04bea.json",
  "parameterValues" : [ ]
}